src/gui/main_screen.py

```diff
@@ -170,15 +172,16 @@ def on_motion(self, event):
             window_height = self.winfo_height()
             x = int((event.x - self.pan_x - max((window_width - self.zoomed_width) // 2, 0)) / self.zoom_level)
             y = int((event.y - self.pan_y - max((window_height - self.zoomed_height) // 2, 0)) / self.zoom_level)
-            coord = (x, y)
-            if not (0 <= x < self.image_array_width and 0 <= y < self.image_array_height):
+            self.tooltip_coord = (x, y)
+
+            if not (0 <= self.tooltip_coord[0] < self.image_array_width and 0 <= self.tooltip_coord[1] < self.image_array_height):
                 self.hide_tooltip()
                 return
-            if coord not in engine.world:
+            if self.tooltip_coord not in engine.world:
```

Comment:
Can't update_tooltip_text() be used here? code is duplicated

Reply:
No it can't because it would bypass code that makes the tooltip appear, update_tooltip_text() only changes what the tooltip shows, not if it appears at all